AI Chat: fix starter assets upload dialog overlap

sanchitmalhotra126 opened the pull request:
<table><tr><td>
<h2> Warning!! </h2>

The [AP CSP Create Performance Task](https://apcentral.collegeboard.org/courses/ap-computer-science-principles/exam) is in progress. The most critical dates are from April 3 - April 30, 2025. Please consider any risk introduced by this PR that could affect our students taking AP CSP. Code.org students taking AP CSP primarily use App Lab for their Create Task, however a small percent use Game Lab. Carefully consider whether your change has any risk of alterering, changing, or breaking anything in these two labs. Even small changes, such as a different button color, are considered significant during this time period. Reach out to the Student Learning team or Curriculum team for more details.
</td></tr></table>

## Summary

Fixes overlap issues with the starter assets dialog in AI Chat. A lot of this was already fixed by https://github.com/code-dot-org/code-dot-org/pull/65309, but there were still a few elements in the header and footer peeking through the overlay. I think there's some bigger issues here with how we handle z-indices across different stacking contexts in the page, but for now, and easy solution is to just use `createPortal` to render the modal on the document body directly. Logically, this also makes some sense as the modal backdrop is meant to cover all content on the page, and so should be at the level of top level view containers like the header and footer.

Before:

<img width="1512" alt="Screenshot 2025-04-21 at 2 14 22 PM" src="https://github.com/user-attachments/assets/597aa541-0af3-463b-8eb6-ea392504460f" />

After:

<img width="1512" alt="Screenshot 2025-04-21 at 2 14 12 PM" src="https://github.com/user-attachments/assets/41cdfe35-21ae-4a08-80fa-f75fea17d30f" />

## Links

- Jira: https://codedotorg.atlassian.net/browse/LABS-1422

## Testing story

Tested locally.

bencodeorg commented:
Does this work well with tab/keyboard navigation?

sanchitmalhotra126 commented:
> Does this work well with tab/keyboard navigation?

Good question, yep seems to work fine/same as before (opening the dialog via keyboard automatically shifts focus into the dialog, and you can tab around).

bencodeorg commented:
Thanks for looking into it! One last q -- does the focus return to the right place when you close the dialog?

sanchitmalhotra126 commented:
> Thanks for looking into it! One last q -- does the focus return to the right place when you close the dialog?

Oh, good question - it does not (it returns to the top of the page, so you have to tab all the way back to the chat window to submit). That being said, I may have found an alternate solution that doesn't require `createPortal` at all - will update!

sanchitmalhotra126 commented:
Updated - instead of using `createPortal` to render the modal on `document.body`, I noticed we were using `position: fixed` for the lab2 container, which if I'm reading [this](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_positioned_layout/Stacking_context) correctly, creates a new "stacking context" for z-ordering elements. By changing the wrapper to use `position: absolute`, the z-index issues go away, and even get fixed for other modals (such as the AI Chat disclaimer modal). As far as I can tell, this has no visual effect on any lab2 labs (checked all labs in the allthethings lab2 showcase and a few other scripts). Tagging @breville and @molly-moen as well. @bencodeorg re: your questions, this also maintains expected behavior for keyboard/tab navigation.

sanchitmalhotra126 commented on apps/src/lab2/views/Lab2Wrapper.module.scss:
```diff
@@ -2,7 +2,7 @@
 .labContainer {
-  position: fixed;
+  position: absolute;
   top: $top-spacing;
```
Looking at this, it's probably more ideal to not use absolute positioning at all, and let the page lay us out wherever we need to be so we don't need to make any assumptions about the size of the header - but that's for a separate change 🙂

molly-moen reviewed (approved):
🎉

bencodeorg reviewed (approved):
Nice!
